src/data/serializer/XYZ.js

```diff
@@ -62,13 +69,33 @@ Ext.define('GeoExt.data.serializer.XYZ', {
             var serialized = {
                 baseURL: source.getUrls()[0],
                 opacity: layer.getOpacity(),
+                imageExtension: this._getImageExtensionFromURL(
+                    source.getUrls()[0]) || 'png',
```

Comment:
[`getUrls()`](http://openlayers.org/en/v3.15.1/apidoc/ol.source.XYZ.html#getUrls) may very well return `null`, we should guard against that case.

Side note: I am not sure if we document enough that we will only really be serializing xyz sources which use the `urls`-config, and sources which use either `url` or `tileUrlFunction` will not produce sth. meaningful.